core/parser.mly

```diff
@@ -929,22 +929,32 @@ forall_datatype:
      S = !(ModuleA.ModuleB.Type).S
 
      Parenthesised versions take priority over non-parenthesised versions.
+
+   jcheney: Modified grammar to allow unparenthesized qualified type names
```

Comment:
Hint: comment "ownership" can be always determined using `git blame`, so no need to sign comments. Especially that they can - and most likely will - be updated by others, which blurs the "ownership".